portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,32 +37,37 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING).addSort("username");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-   
-    List<String> commentsStored = new ArrayList<String>();
+    List<Comment> commentsStored = new ArrayList<Comment>();
+    Gson gson  = new Gson();
     for (Entity entity : results.asIterable()) {
-      String comment = (String) entity.getProperty("comment");
+      Comment comment = new Comment((String)entity.getProperty("username"), (String)entity.getProperty("comment"));
       commentsStored.add(comment);
     }
-
-    Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(commentsStored));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = request.getParameter("text-input");
-    comments.add(comment);
-    long timestamp = System.currentTimeMillis();
-    Entity commentEntity = new Entity("Comments");
-    commentEntity.setProperty("comment", comment);
-    commentEntity.setProperty("timestamp", timestamp);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
-    response.sendRedirect("/index.html");
-    
+    //String comment = request.getParameter("text-input");
```

Comment:
Here and in some other files it looks like there's a bit of commented out code that you can delete.